scripts/publish-package.js

```diff
@@ -38,25 +42,39 @@ async function publishPackage(packageName, client, keypair) {
     });
 
     // Transfer the upgrade capability to the sender so they can upgrade the package later if they want.
+    let address;
+
+    if (!options.sender) {
+        address = keypair.getPublicKey().toSuiAddress();
+    } else {
+        address = options.sender;
+    }
+
     tx.transferObjects([cap], tx.pure(address));
 
-    const publishTxn = await client.signAndExecuteTransactionBlock({
-        transactionBlock: tx,
-        signer: keypair,
-        options: {
-            showEffects: true,
-            showObjectChanges: true,
-            showContent: true,
-        },
-        requestType: 'WaitForLocalExecution',
-    });
-    if (publishTxn.effects?.status.status !== 'success') throw new Error('Publish Tx failed');
+    if (!options.offline) {
```

Comment:
what's the use case for offline support for publishing packages? similar to evm, i don't see why we're offline signing deployments as opposed to other txs. I'm ok to add it since it's a small change and it's already ready, but otherwise best to avoid spending time on use cases we don't need